### DISCUSSION TITLE
Inserting a gif into a png image

### DISCUSSION THREAD

**Discussion by KubaZ2** [dlemstra/Magick.NET#958] [Help]
Hello, I want to insert a gif into an image. I have made it this way:
```cs
// `avatar` is the gif, `image` is the png image
for (int i = 0; i < avatar.Count; i++)
{
    IMagickImage<byte> avatarFrame = avatar[i];
    var newImage = image.Clone();
    newImage.AnimationDelay = avatarFrame.AnimationDelay;
    newImage.AnimationIterations = avatarFrame.AnimationIterations;
    newImage.Composite(avatarFrame, 18, 18, CompositeOperator.Over);
    avatar[i] = newImage;
}
```
but I don't think it is good (it takes a lot of time).

Do you have any good idea?

**Comment by dlemstra** [comment]
This is the way that you should do this. But you probably get a small performance improvement if you don't `.Clone` the image.

p.s. If you clone the image and overwrite the original in the collection it is your responsibility to dispose the original image.

**Reply by KubaZ2** [reply]
If I remove `.Clone` Magick.net throws `InvalidOperationException` `Not allowed to add the same image to the collection.`

**Reply by dlemstra** [reply]
You don't need to add it again.

**Reply by KubaZ2** [reply]
But I have only removed `.Clone()` from code.

**Reply by KubaZ2** [reply]
```cs
                            for (int i = 0; i < avatar.Count; i++)
                            {
                                IMagickImage<byte> avatarFrame = avatar[i];
                                var newImage = image;
                                newImage.AnimationDelay = avatarFrame.AnimationDelay;
                                newImage.AnimationIterations = avatarFrame.AnimationIterations;
                                newImage.Composite(avatarFrame, 18, 18, CompositeOperator.Over);
                                avatar[i] = newImage;
                            }
```

**Reply by dlemstra** [reply]
I misread your code. You do require the `Clone` when you want to reuse the same `image`. I don't know what you could to to reduce the time of your operation.
